src/components/views/dialogs/keybackup/RestoreKeyBackupDialog.js

```diff
@@ -0,0 +1,172 @@
+            this.setState({
+                loading: false,
+                restoreError: e,
+            });
+        }
+    },
+
+    _loadBackupStatus: async function() {
+        this.setState({
+            loading: true,
+            loadError: null,
+        });
+        try {
+            const backupInfo = await MatrixClientPeg.get().getKeyBackupVersion();
+            this.setState({
+                loadError: null,
+                loading: false,
+                backupInfo,
+            });
+        } catch (e) {
+            console.log("Error loading backup status", e);
+            this.setState({
+                loadError: e,
+                loading: false,
+            });
+        }
+    },
+
+    render: function() {
+        const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
+        const Spinner = sdk.getComponent("elements.Spinner");
+
+        let content;
+        if (this.state.loading) {
+            content = <Spinner />;
+        } else if (this.state.loadError) {
+            content = _t("Unable to load backup status");
+        } else if (this.state.restoreError) {
+            content = _t("Unable to restore backup");
+        } else if (this.state.backupInfo === null) {
+            content = _t("No backup found!");
+        } else if (this.state.recoverInfo) {
+            let failedToDecrypt;
+            if (this.state.recoverInfo.total > this.state.recoverInfo.imported) {
+                failedToDecrypt = <p>{_t(
+                    "Failed to decrypt %(failedCount)s sessions!",
+                    {failedCount: this.state.recoverInfo.total - this.state.recoverInfo.imported},
+                )}</p>;
+            }
+            content = <div>
+                <p>{_t("Restored %(sessionCount)s session keys", {sessionCount: this.state.recoverInfo.imported})}</p>
+                {failedToDecrypt}
+            </div>;
+        } else {
+            const DialogButtons = sdk.getComponent('views.elements.DialogButtons');
+
+            let keyStatus;
+            if (this.state.recoveryKey.length === 0) {
+                keyStatus = <div className="mx_RestoreKeyBackupDialog_keyStatus"></div>;
+            } else if (this.state.recoveryKeyValid) {
+                keyStatus = <div className="mx_RestoreKeyBackupDialog_keyStatus">
+                    {"\uD83D\uDC4D "}{_t("This looks like a valid recovery key!")}
+                </div>;
+            } else {
+                keyStatus = <div className="mx_RestoreKeyBackupDialog_keyStatus">
+                    {"\uD83D\uDC4E "}{_t("Not a valid recovery key")}
+                </div>;
+            }
+
+            content = <div>
+                {_t("Please enter the recovery key generated when you set up key backup")}<br />
+                <textarea
+                    onChange={this._onRecoveryKeyChange}
+                    value={this.state.recoveryKey}
+                    style={{width: "90%"}}
```

Comment:
inline style?! 😿🐼

Reply:
I'm assuming the CSS stuff will definitely go in the bin soon, so didn't bother pulling it out to CSS.